src/main/java/com/hw/bookstore/dto/BookRequestDto.java

```diff
@@ -0,0 +1,32 @@
+package com.hw.bookstore.dto;
+
+import jakarta.validation.constraints.DecimalMax;
+import jakarta.validation.constraints.DecimalMin;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import java.math.BigDecimal;
+
+public record BookRequestDto(
+
+        @NotNull(message = "Title can not be null")

```

Comment:
Better use NotBlank with literals. Here you can find info about the difference https://www.baeldung.com/java-bean-validation-not-null-empty-blank

Reply:
Yes, it's a great comment. But I've added size validation and it already checks for empty, so we just need to check for null.

Reply:
annotation size will not validate a value like "     "